docs/_docs/ssl-tls.adoc

```diff
@@ -0,0 +1,193 @@
+= SSL/TLS
+
+This page explains how to configure SSL/TLS encryption between the cluster nodes (server and client) and the clients that connect to your cluster.
+
+== Considerations
+
+All internal connections in the cluster context, as well as cluster's user interaction interfaces, are SSL-enabled. The communication categories are as follows: 
+
+* Between the user and the cluster (node): REST
+* Between the user and the platform clients:
+
+** CLI
+** Client
+
+* Between nodes: Network (Messaging, Scalecube)
+
+All SSL configurations activities are performed at the node level.
+
+Due to Micronaut limitations, Apache Ignite does nopt support direct paths to SSL certificates. Instead, it utilizes PKCS12 and JKS keystore.
+//Need a link to Micronaut in general; if possible, to the limitation in particular 
+
+== REST
+
+The standard implementation of SSL support for REST involves configuring a secure connection on a separate port. Apache Ignite also supports the DUAL protocol feature.
+//Need a link to a description of the DUAL protocol. 
+
+To configure SSL in Micronaut, all the necessary properties must be defined in the Micronaut notation. Map the Apache Ignite 3.x REST security configuration to the Micronaut one:
+
+[source,json]
+----
+"rest": {
+    "dualProtocol": false,
+    "httpToHttpsRedirection": false,
+    "ssl": {
+        "enabled": false,
+        "port": 10400,
+        "portRange": 100,
+        "keyStore": {
+            "type": "PKCS12",
+            "path": "must not be empty",
+            "password": "may be empty"
+        }
+    }
+}
+----
+
+== Clients and JDBC
+
+Apache Ignite 3.x Client implementation is based on the Netty framework, which supports configuration for security connections via SSLContextBuilder.
+
+=== Client-side Configuration
+
+[source,java]
+----
+SslContextBuilder sslBuilder = SslContextBuilder
+   .forClient()
+   .keyManager(getKeyManagerFactory())
+   .trustManager(getTrustManagerFactory());
+}
+----
+
+=== Server-side Configuration
+
+[source,java]
+----
+SslContextBuilder builder = SslContextBuilder.forServer(certChainInput, keyInput)
+    .ciphers(getCiphers(), getCiphersFilter())
+    .sessionTimeout(serverSslConfig.getSessionTimeout())
+    .sslProvider(sslProvider);                 
+.trustManager(trustedCerts.toArray(new X509Certificate[0]))
+    .clientAuth(serverSslConfig.getClientAuth());
+----
+
+View the link:https://github.com/devsunny/netty-ssl-example/blob/master/src/main/java/com/asksunny/ssl/SecureSocketSslContextFactory.java[Netty SSL configuration].
```

Comment:
I am not sure that this brings any peace of information for users. As well as code examples. It is just internal implementation.